### ISSUE TITLE
Cannot install via opam on MacOS

### ISSUE THREAD

**Issue by ose20** [garrigue/labltk#13]
While trying to install labltk via opam, I received this error message (Some excerpts) :
```
<><> Processing actions <><><><><><><><><><><><><><><><><><><><><><><><><><>  🐫
[ERROR] The compilation of labltk failed at "/Users/myname/.opam/opam-init/hooks/sandbox.sh build make library
        opt".

#=== ERROR while compiling labltk.8.06.9 ======================================#
# context     2.0.7 | macos/x86_64 | ocaml-base-compiler.4.10.0 | https://opam.ocaml.org#2967e1c2
# path        ~/.opam/default/.opam-switch/build/labltk.8.06.9

# /usr/local/include/tk.h:96:13: fatal error: 'X11/Xlib.h' file not found
# #   include <X11/Xlib.h>
#             ^~~~~~~~~~~~
# 1 error generated.
# make[1]: *** [cltkCaml.o] Error 1
# make: *** [library] Error 2

<><> labltk.8.06.9 troubleshooting ><><><><><><><><><><><><><><><><><><><><>  🐫
=> This package requires Tcl/Tk with its development packages installed on your system
```

There are two things I don't understand about this message.

Firstly, this message said that I need to install Tcl/Tk, but I have already installed them from (https://www.activestate.com/products/tcl/downloads/). 
So, I have Tcl(and Tk).framework in /Library/Frameworks. And their version are 8.6.
Why opam does not recognize them ?

Secondly, this message said that X11/Xlib.h file was not found, but I have this file.
For example, 
```
/usr/local/Cellar/tcl-tk/8.6.10/lib/tcl8.6/critcl3.1.18/critcl_c/tcl8.6/X11/Xlib.h
/Library/Frameworks/Tcl.framework/Versions/8.6/Resources/Scripts/critcl3.1.17/critcl_c/tcl8.6/X11/Xlib.h
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/System/Library/Frameworks/Tk.framework/Versions/8.5/Headers/X11/Xlib.h
/opt/X11/include/X11/Xlib.h
etc
```
Similarly, why opam does not recognize these files ?

This is the first question I'm asking on github. So I apologize if this is an inappropriate question.

**Comment by rsk0315** [comment]
TL;DR: try to do

```
$ sudo ln -s /opt/X11 /usr/X11
```

and then

```
$ opam install labltk
```

---

Here is what I did.

I tried to install this with:

```
$ brew install opam
$ eval $(opam env)
$ opam install labltk
```

and got the same error as above. Then I installed [XQuartz](https://www.xquartz.org/) and got:

```
-> installed labltk.8.06.9
Done.
```

However I noticed that the installation of XQuartz made `/opt/X11/include/X11/Xlib.h`, which you already have.
Then I remembered that I had got the same issue (missing `X11/Xlib.h`) before in other project. At that time, I created a symbolic link (`/opt/X11 -> /usr/X11`) in order to let the installer detect them. So just installation of XQuartz made it for me this time.

I hope this helps.

**Comment by ose20** [comment]
I was able to solve the problem.
In my case,  
```
$ sudo ln -s /opt/X11 /usr/X11
```
resulted in
```
ln: /usr/X11: Operation not permitted
```
However just installation of XQuartz was enough also for me.
Anyway, thank you for taking the time to teach me.
